src/platform/site-wide/user-nav/containers/Main.jsx

```diff
@@ -42,7 +49,8 @@ export class Main extends React.Component {
       window.location.replace(redirectPath);
     }
 
-    const shouldCloseLoginModal = currentlyLoggedIn && showModal;
+    const shouldCloseLoginModal =
+      !prevProps.currentlyLoggedIn && currentlyLoggedIn && showLoginModal;
```

Comment:
We originally had this, but this was causing some problems where the modal would occasionally not close after `currentlyLoggedIn` became `true` for some reason.

We decided we would always want to make sure that the modal is not appearing when the user is logged in anyway, so it wouldn't be a big issue to remove the `!prevProps.currentlyLoggedIn` part of the condition. I don't anticipate this condition being triggered more times than it already would with the `prevProps` being involved, and it guarantees the modal being closed when the user is logged in.